atlp-rwanda/code-ninjas-be
# #181414755 Ft user registration

# #181414755 On successful registration, users should receive a JWT

### What does this PR do?
Register a user and generate a jwt token when an account is successful created.
### Description of Task to be completed?
Once a Users is registered via the registration endpoint, receives a JWT to be required on all subsequent calls to all other endpoints that require authentication.

The User receives a descriptive error message when they enter invalid parameters in an attempt to signup.
if a user has an 
existing user email,
invalid email or password, he is prompted with the appropriate error message. on the other hand the 
password of length less than 8, and a password must contain an Upper case, a Number, a special character to be recognised as strong.

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to develop branch
  - ``yarn install``: to install all dependencies
  - ``yarn dev``: to start the development server
  -   Open postman
  - Navigate to ``http://localhost:3000/api/v2/users/register``: use ``POST`` http verb  to input the data from the body to be sent to the server example:``{
    "FirstName": "Jane",
    "LastName": "Doe",
    "Email": "test@gmail.com",
    "UserName" : "Doe",
    "Password" : "Password@2022"
}``
  - Create .env file
  - add database url using "DB_CONNECT" as variable
- Navigate to heroku page of the team and open Heroku Postgress DB go to the settings and check the URI of the DB paste it in  "DB_CONNECT" variable.

### What are the relevant pivotal tracker stories?
[#181414755](https://www.pivotaltracker.com/story/show/181414755)
